### ISSUE TITLE
Fix request: false positive for OptimizeMapOrSetForEnum

### ISSUE THREAD

**Issue by ntinnemeier** [jborgers/PMD-jPinpoint-rules#166]
The OptimizeMapOrSetForEnum rule triggers on the following code, but shouldn't:

```
public class GreetingsViolation {
    enum Language {
        NL, EN, FR;
    }

    private static final Map<Language, String> greetings = new EnumMap<>(Language.class);

    static {
        greetings.put(NL, "hallo");
        greetings.put(EN, "hello");
        greetings.put(FR, "bonjour");
    }

    public static void main(String[] args) {
        Map<Language, String> greetings = GreetingsViolation.greetings;
    }
}
```

Note that when we change the type of the `greetings` from `Map` map to `EnumMap` no violation of the OptimizeMapOrSetForEnum rule is reported.

**Comment by jborgers** [comment]
Might be solved by #194 , please retest

**Comment by ntinnemeier** [comment]
I see #194 got reopened, so I'll wait with testing.

**Comment by jborgers** [comment]
It is solved better now

**Comment by ntinnemeier** [comment]
Works as a charm. I also verified that the rule still triggers in case there actually is a violation (using a HashMap instead of EnumMap) and it does.
